ds3-sdk/src/main/java/com/spectralogic/ds3client/metadata/MetadataReceivedListenerImpl.java

```diff
@@ -0,0 +1,254 @@
+                        permissions.add(AclEntryPermission.DELETE);
+                        break;
+                    case "10":
+                        permissions.add(AclEntryPermission.READ_ACL);
+                        break;
+                    case "11":
+                        permissions.add(AclEntryPermission.WRITE_ACL);
+                        break;
+                    case "12":
+                        permissions.add(AclEntryPermission.WRITE_OWNER);
+                        break;
+                    case "13":
+                        permissions.add(AclEntryPermission.SYNCHRONIZE);
+                        break;
+                }
+            }
+            builderWindow.setPrincipal(bRiceUser);
+            builderWindow.setPermissions(permissions);
+            builderWindow.setType(AclEntryType.ALLOW);
+            final AclEntry aclEntry = builderWindow.build();
+            aclEntries.add(aclEntry);
+        } catch (final Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    //set the last modified time saved on server
+    private void setModifiedTime(final String filePath, final Long modifiedTime) {
+        try {
+            final Path file = Paths.get(filePath);
+            final BasicFileAttributes attr =
+                    Files.readAttributes(file, BasicFileAttributes.class);
+            final FileTime ft = FileTime.fromMillis(modifiedTime);
+            Files.setLastModifiedTime(file, ft);
+        } catch (final Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    //set the permissions saved on server
+    private void setPermissions(final String filePath, final String permissions) {
+        try {
+            final Path file = Paths.get(filePath);
+            final Set<PosixFilePermission> perms =
+                    PosixFilePermissions.fromString(permissions);
+/*            FileAttribute<Set<PosixFilePermission>> attr =
+                    PosixFilePermissions.asFileAttribute(perms);*/
+            Files.setPosixFilePermissions(file, perms);
+        } catch (final Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    //set owner and group name on local from the black perl server
+    private void setOwnerNGroup(final String filePath, final String ownerName, final String groupName) {
+        try {
+            final Path file = Paths.get(filePath);
+            final UserPrincipal owner = file.getFileSystem().getUserPrincipalLookupService()
+                    .lookupPrincipalByName(ownerName);
+            Files.setOwner(file, owner);
+            final GroupPrincipal group =
+                    file.getFileSystem().getUserPrincipalLookupService()
+                            .lookupPrincipalByGroupName(groupName);
+            Files.getFileAttributeView(file, PosixFileAttributeView.class)
+                    .setGroup(group);
+        } catch (final Exception e) {
+            e.printStackTrace();
```

Comment:
Exceptions should not be handled in this way. At a minimum it should be logged with a slf4j logger, or wrapped in a runtime exception and re-thrown.